on-premise-docker-registry/README.md

```diff
@@ -0,0 +1,191 @@
+  ![cc-config](assets/cc-config.png)
+
+## nginx as reverse proxy
+
+We use Nginx as a reverse proxy to forward the HTTP requests for pulling Docker images from the on-premise Docker registry using Connectivity Proxy and Cloud Connector.
+
+It is exposed as the `NodePort` service. This exposes the Service on the Kubernetes worker node on a port. Kubelet is a component running on each Kubernetes worker node. Among other tasks, it is responsible for pulling the Docker images.
+
+When creating a deployment, we specify the Docker registry as `localhost:{NodePort}`. This is the address of the nginx reverse proxy. The nginx reverse proxy then forwards the call to the on-premise Docker registry using Connectivity Proxy and Cloud Connector.
+
+This sample shows a simple configuration with nginx as a reverse proxy. You can use any other reverse proxy implementation based on your on-premise Docker registry behavior and APIs.
+
+- Create namespace and enable istio sidecar injection if not done earlier.
+
+```shell
+kubectl create namespace ${NAMESPACE}
+kubectl label namespace ${NAMESPACE} istio-injection=enabled
+```
+
+- Deploy the nginx as a reverse proxy. The following components will be deployed:
+  - [ConfigMap](./k8s/configmap.yaml) for nginx configuration
+  - [Deployment and Service](./k8s/deployment.yaml)
+  - [PeerAuthentication](./k8s/peer-authentication.yaml) set to `PERMISSIVE` to allow for communication between kubelet and the nginx reverse proxy.
+
+```shell
+make deploy-nginx-reverse-proxy
+```
+
+- Wait for the nginx reverse proxy to be up and running
+
+```shell
+make check-nginx-reverse-proxy
+```
+
+- Export NodePort for the nginx reverse proxy as an environment variable
+
+```shell
+export NGINX_NODE_PORT=$(kubectl get svc nginx -o jsonpath='{.spec.ports[0].nodePort}')
+```
+
+## Test workload
+
+Let's deploy a sample workload that will use an image from the on-premise Docker registry.
+
+- Log in to the created on-premise docker registry
+
+```shell
+make docker-login
+```
+
+- Create and store a sample Docker image in the on-premise Docker registry. It uses an image tag based on the current time.
+
+```shell
+make create-test-image
+```
+
+- Update the [deployment.yaml for test workload](./test-image-deployment/deployment.yaml).
+  - Replace `{nginx-reverse-proxy-node-port}` with the `NGINX_NODE_PORT` value.
+  - Replace `{generate-image-tag}` with the tag of the generated image.
+
+- Create Kubernetes Secret with credentials to pull the Docker image.
+
+```shell
+make create-secret-to-pull-image
+```
+
+- Deploy the test workload. Notice that in the [deployment file](./test-image-deployment/deployment.yaml), the image is specified as `localhost:30930/....`. This is the localhost wrt to the Kubernetes worker node, and the port is the NodePort of the nginx reverse proxy service.
+
+```shell
+make deploy-test-workload
+```
+
+- Wait until the Pod is up and running. Check the status using the following command:
+
+```shell
```

Comment:
add 3 blank spaces at the beginning of each line of the code block for proper indentation below the list item.